spec/iglu-client/schema_key_spec.rb

```diff
@@ -38,15 +38,31 @@
                '   "deviceid": "asdfasdfasdfasdfcwer234fa$#ds±f324joa"' \
                ' }' \
                '}'
-    expect(Iglu::SelfDescribingJson.parse_json(JSON.parse(instance)).valid?(@resolver)) == false
+    expect(Iglu::SelfDescribingJson.parse_json(JSON.parse(instance)).valid?(@resolver)).to eq(false)
   end
 
   it 'correctly parses Iglu URI into object' do
-    expect(Iglu::SchemaKey.parse_key("iglu:com.snowplowanalytics.snowplow/event/jsonschema/1-0-1")) == Iglu::SchemaKey.new("com.snowplowanalytics.snowplow", "event", "jsonschema", Iglu::SchemaVer.new(1, 0, 1))
+    expect(Iglu::SchemaKey.parse_key("iglu:com.snowplowanalytics.snowplow/event/jsonschema/1-0-1")).to eq(Iglu::SchemaKey.new("com.snowplowanalytics.snowplow", "event", "jsonschema", Iglu::SchemaVer.new(1, 0, 1)))
   end
 
   it 'correctly parses SchemaVer into object' do
-    expect(Iglu::SchemaVer.parse_schemaver("2-0-3")) == Iglu::SchemaVer.new(2, 0, 3)
+    expect(Iglu::SchemaVer.parse_schemaver("2-0-3")).to eq(Iglu::SchemaVer.new(2, 0, 3))
+  end
+
+  it 'throws exception on an incorrect SchemaVer (0 based versioning)' do
+    expect { Iglu::SchemaVer.parse_schemaver("0-1-2") }.to raise_error(Iglu::IgluError)
+  end
+
+  it 'throws exception on an incorrect SchemaVer (with lower case letters)' do
+    expect { Iglu::SchemaVer.parse_schemaver("a-b-c") }.to raise_error(Iglu::IgluError)
+  end
+
+  it 'throws exception on an incorrect SchemaVer (with upper case letters)' do
+    expect { Iglu::SchemaVer.parse_schemaver("A-B-C") }.to raise_error(Iglu::IgluError)
+  end
+
+  it 'throws exception on an incorrect SchemaVer (dot formatted)' do
+    expect { Iglu::SchemaVer.parse_schemaver("2.0.3") }.to raise_error(Iglu::IgluError)
```

Comment:
Let's also add few expectations for multiple-digits versions. They're correct, e.g. `10-0-11` should be correct, `10-a-1` should be not correct.